### ISSUE TITLE
ExponentOfPrime vs PValuation

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/utils#16]
For your information, `ExponentOfPrime` in utils does the same as `PValuation` in GAP; the latter is faster.
```
gap> SetX([-100..100], Primes, {n,p} -> PValuation(n,p) = ExponentOfPrime(n,p));
[ true ]
gap> for n in [1..10^4] do for p in Primes2 do PValuation(n,p); od; od; time;
1754
gap> for n in [1..10^4] do for p in Primes2 do ExponentOfPrime(n,p); od; od; time;
2561
```

**Comment by cdwensley** [comment]
Thanks - a PR noting this function as obsolete (liable to be removed in due course) is currently being checked.  As this function comes from RCWA I will make a PR changing it to PValuation. 
The ResClasses package does not appear to use ExponentOfPrime. 
Is there any straightforward way to find out if it is used by any other package?

**Comment by fingolfin** [comment]
What I do is: I use a recursive `grep` (Or rather, this: 
https://github.com/BurntSushi/ripgrep) to search through my pkg directory (one with the content of the latest package archive; the other with all packages from GitHub, maintained using these scripts: 
https://github.com/fingolfin/gap-pkg-repos)

**Comment by alex-konovalov** [comment]
I have an alias `grepgap` in my ~/.bash_profile

    alias grepgap='grep -n -R --include=*.g* '

for similar purposes.

**Comment by cdwensley** [comment]
Thanks - more straightforward than having to install other applications.  This gives: 
ChristoleysiMac:pkg christopherwensley$ grep -n -R --include=*.g* ExponentOfPrime ./*
./rcwa-4.6.2/lib/rcwagrp.gi:3471:    powers_impossible := List(primes_onlymod,p->p^(ExponentOfPrime(m,p)+1));
./rcwa-4.6.2/lib/rcwagrp.gi:6022:      powers_impossible := List(primes_onlymod,p->p^(ExponentOfPrime(m,p)+1));
./utils-0.57/lib/number.gd:37:#F  ExponentOfPrime( <n>, <p> )
./utils-0.57/lib/number.gd:41:DeclareGlobalFunction( "ExponentOfPrime" );
./utils-0.57/lib/number.gi:47:#F  ExponentOfPrime( <n>, <p> )
./utils-0.57/lib/number.gi:49:InstallGlobalFunction( ExponentOfPrime,
while including *.tst and *.xml only finds entries in Utils.

**Comment by cdwensley** [comment]
PR #6 in RCWA changes the only two occurrences of ExponentOfPrime to PValuation.

**Comment by alex-konovalov** [comment]
Good. Two remarks:
- sometimes worth also to check .tst or .xml files
- `#6` is rendered as a link to number 6 in Utils. If you want a proper cross-reference for the ease of further navigation, better to paste full URL https://github.com/gap-packages/rcwa/pull/6 of use `[#6](https://github.com/gap-packages/rcwa/pull/6)` which result in [#6](https://github.com/gap-packages/rcwa/pull/6).

**Comment by cdwensley** [comment]
Remember to delete ExponentOfPrime from Utils once the RCWA PR is merged.

**Comment by cdwensley** [comment]
Since RCWA-4.6.3 has been released, with the PR merged, Utils-0.61 is now released with ExponentOfPrime obsolete, but listed in the manual chapter on obsolete functions.
